### PR TITLE
Initial commit of audio/I2S driver without any BSP support for any…

### PR DESCRIPTION
**Pull request by btheosam**
… hardware

### CODE REVIEW

File: os/include/tinyara/audio/i2s.h [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+ * Input Parameters:
+ *   dev  - Device-specific state data
+ *   rate - The I2S sample rate in samples (not bits) per second
+ *
+ * Returned Value:
+ *   Returns the resulting bitrate
+ *
+ ****************************************************************************/
+
+#define I2S_RXSAMPLERATE(d,f) ((d)->ops->i2s_rxsamplerate(d,r))
```
Comment (sunghan-chang) [inline os/include/tinyara/audio/i2s.h]: You should add space after comma. So let's change (d,f) -> (d, f).
There are many at belows.

File: os/include/tinyara/audio/i2s.h [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+struct i2s_dev_s;
+typedef CODE void (*i2s_callback_t)(FAR struct i2s_dev_s *dev, FAR struct ap_buffer_s *apb, FAR void *arg, int result);
+
+/* The I2S vtable */
+
+struct i2s_ops_s {
+	/* Receiver methods */
+
+	CODE uint32_t(*i2s_rxsamplerate)(FAR struct i2s_dev_s *dev, uint32_t rate);
```
Comment (sunghan-chang) [inline os/include/tinyara/audio/i2s.h]: You need to add space between return type and function.

File: os/include/tinyara/audio/tone.h [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+#ifndef __INCLUDE_TINYARA_AUDIO_TONE_H
+#define __INCLUDE_TINYARA_AUDIO_TONE_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/compiler.h>
```
Comment (sunghan-chang) [inline os/include/tinyara/audio/tone.h]: Why do we need this header?

File: os/include/tinyara/audio/i2s.h (hunk 1 above)
Comment (btheosam) [inline os/include/tinyara/audio/i2s.h]: OKay, will address this.

File: os/include/tinyara/audio/i2s.h (hunk 2 above)
Comment (btheosam) [inline os/include/tinyara/audio/i2s.h]: Okay, done

File: os/include/tinyara/audio/tone.h (hunk 3 above)
Comment (btheosam) [inline os/include/tinyara/audio/tone.h]: for __cplusplus

**Comment by btheosam** [comment]
Has been addressed in another commit.

**Pull request closed without merging**